docs/DeveloperGuide.md

```diff
@@ -154,6 +154,104 @@ Classes used by multiple components are in the `seedu.linkedout.commons` package
 
 This section describes some noteworthy details on how certain features are implemented.
 
+### Edit feature
+
+#### Implementation
+
+The proposed edit mechanism is facilitated by `EditApplicantDescriptor`. `EditApplicantDescriptor` stores the details of the applicant to change.
+
+The user need to specify an `Index` to select the applicant to edit. `EditCommand` extends `Command` and implements the `Command#execute()` method.
+
+Given below is an example usage scenario of how an applicant is edited.
+
+1. The user enters the edit command with the specific fields to edit, `edit 1 r/HR Interview`.
+2. LinkedOUT updates the applicant with the edited information.
+
+The following sequence diagram shows how the edit operation works:
+
+![EditSequenceDiagram](images/EditSequenceDiagram.png)
+
+#### Design considerations:
+
+**Aspect: How edit executes:**
+
+* **Alternative 1 (current choice):** Creates a new applicant replace old information.
+    * Pros: Easy to implement.
+    * Cons: May have performance issues in terms of memory usage.
+
+* **Alternative 2:** Replace individual fields inside original applicant
+    * Pros: Will use less memory (Do not have to create an extra applicant).
+    * Cons: We must ensure that the implementation of each individual command to change an information is correct.
+    
+
+### View feature
+
+#### Rationale
+
+The view feature searches a **single** applicant in the application and returns the applicant's details. 
+It is used when users wish to find a specific user they have in mind.
+It takes in a single case-insensitive parameter, which is the applicant's name. No prefix is required.
```

Comment:
I like how you specified **single**!! Maybe can change applicant's name to applicant's full name instead

Reply:
Thanks for the spot, updated!